### DISCUSSION TITLE
Syncing slash commands with their respective normal (prefix-based) commands

### DISCUSSION THREAD

**Discussion by ghost** [Rapptz/discord.py#9105] [Q&A]
I probably used the incorrect terminology in regards to syncing them. But I have a beg command I made and it has both a prefix-based command and a slash command. 

Error handler:
`if isinstance(error, commands.CommandOnCooldown):`
`      embed = discord.Embed(title = '', color = discord.Color.red())`
`      embed.add_field(name = "Error", value = "Still on cooldown, please try again in {:.2f}s'.format(error.retry_after))`
`      await ctx.send(embed = embed)`

I know that I will likely need to add another one of these^ for slash commands.

Then I have:
`@commandscooldown(1, 240, commands.BucketType.user)`
On the prefix-based command.

Following the documentation on slash command cooldowns: https://discordpy.readthedocs.io/en/latest/interactions/api.html?highlight=cooldown#discord.app_commands.checks.cooldown

I figure I could use something like what's shown, but prior to digging down a rabbit hole, how would I make it so that if a user types the prefix based command, they won't be able to then also use the slash command until a "shared" cooldown so-to-speak ends?

**Comment by Rapptz** [comment]
If you use a hybrid command then the cooldowns and checks etc are all shared and work transparently as expected.

**Reply by Rapptz** [reply]
Everything works mostly the same with a few restrictions, like `Union` types not working or `*args` syntax.
